ehcache-impl/src/main/java/org/ehcache/impl/internal/store/heap/OnHeapStore.java

```diff
@@ -231,11 +228,11 @@ public OnHeapStore(Configuration<K, V> config, TimeSource timeSource, Copier<K>
     this.storeEventDispatcher = eventDispatcher;
 
     if (keyCopier instanceof IdentityCopier) {
-      this.map = new SimpleBackend<>(byteSized, castBackend(backingMapSupplier));
+      this.backendSupplier = () -> new SimpleBackend<>(byteSized);
     } else {
-      this.map = new KeyCopyBackend<>(byteSized, keyCopier, castBackend(backingMapSupplier));
```

Comment:
I think the `castBackend(...)` method should be unused now.

Reply:
Yes, I have removed it now.